zerver/models/messages.py

```diff
@@ -648,6 +648,15 @@ def get_usermessage_by_message_id(user_profile: UserProfile, message_id: int) ->
         return None
 
 
+class IdempotentMessage(models.Model):
+    # key has the following format: realm_id:user.id:<queue_id>:local_id
+    key = models.CharField(max_length=150, db_index=True, unique=True)
+    time_sent = models.DateTimeField(
+        default=timezone_now,
+        db_index=True,
+    )
```

Comment:
This table name is confusing. Maybe something like `UniqueMessageSendRequest` would be more clear? Also, your comments/docstrings should explain what it's for / what it does, not just the key format.

(Though see elsewhere where I suggest a different key scheme).